### ISSUE TITLE
OptionsManager needs Options Provider keys list.

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#169]
Sketchup::OptionsManager needs Options Provider keys list in Overview, and also perhaps in `[]` method.

**Comment by mariocha** [comment]
_Output all options available_
`options_manager = Sketchup.active_model.options`
`options_manager.keys.each { |options_provider|`
`  puts options_provider.name`
`  options_provider.each { |key, value|`
`    puts "> #{key} - #{value}"`
`  }`
`}`
The code above in the documentation will yield a list if you delete  **keys**  form 'options_manager.keys.each'

**Comment by DanRathbun** [comment]
I know this, and also reported the example code error already.
The documentation still needs the list (The C API documentation has it. The Ruby API docs need to "follow suit".)

**Comment by mariocha** [comment]
yeah, I expected you to already know. Just wanted to post for others who might stumble on this like I did before figuring it out.

**Comment by DanRathbun** [comment]
Yea ... 3 years and still this issue has not been dealt with.

**Comment by Eneroth3** [comment]
Broken code example tracked by #204.

Regarding they keys, I think it's best to list them in one place to avoid them being out of sync when updating. DRY.

However, just a list doesn't add much more than what you can already get from querying the keys from the API. I'm wondering if it isn't better to add a separate file to the docs listing the options with a sentence of what they represent and/or where in the UI they are controlled.

**Comment by DanRathbun** [comment]
> I'm wondering if it isn't better to add a separate file to the docs listing the options with a sentence of what they represent and/or where in the UI they are controlled.

**I disagree.** Coders need to have them where they will go looking for them.

They need to be where they need to be. And what makes the Ruby API different from the C API where they **are listed** ?

(It's basically a copy and paste task from the C API code to the Ruby binding code.)

For [`OptionsManager`](https://ruby.sketchup.com/Sketchup/OptionsManager.html) class ...
The names of providers are actually the valid arguments to the [`#[]`](https://ruby.sketchup.com/Sketchup/OptionsManager.html#[]-instance_method) method, and so **should** be listed here.

For [`OptionsProvider`](https://ruby.sketchup.com/Sketchup/OptionsProvider.html) class ...
There is only a partial listing of keys ... ie those added in recent versions.
And again, the keys are the valid arguments to the [`#[]`](https://ruby.sketchup.com/Sketchup/OptionsProvider.html#[]-instance_method) and [`#[]=`](https://ruby.sketchup.com/Sketchup/OptionsProvider.html#[]=-instance_method) methods, and so **should** be listed here.

----

Also ... theoretically, these classes could be used to expose non-model options. Ie, I've discussed in the past that we could get readonly application level options manager and providers that use these classes.

Even better would be to expose these to extension developers as objects they can instantiate for plugin options.

**Comment by DanRathbun** [comment]
> Coders need to have them where they will go looking for them.

A perfect example: These keys **are** listed but **not** where we go looking for them.

I just stumbled across their listing in the [`OptionsProviderObserver`](https://ruby.sketchup.com/Sketchup/OptionsProviderObserver.html) class docstring.
